lib/src/time.cpp

```diff
@@ -11,55 +11,55 @@ uint64_t mxlGetTime()
 
 extern "C"
 MXL_EXPORT
-uint64_t mxlGetCurrentHeadIndex(Rational const* editRate)
+uint64_t mxlGetCurrentIndex(Rational const* editRate)
 {
     if ((editRate == nullptr) || (editRate->denominator == 0) || (editRate->numerator == 0))
     {
         return MXL_UNDEFINED_INDEX;
     }
 
     auto const now = currentTime(mxl::lib::Clock::TAI);
-    return now ? mxlTimestampToHeadIndex(editRate, now.value) : MXL_UNDEFINED_INDEX;
+    return now ? mxlTimestampToIndex(editRate, now.value) : MXL_UNDEFINED_INDEX;
 }
 
 extern "C"
 MXL_EXPORT
-uint64_t mxlTimestampToHeadIndex(Rational const* editRate, uint64_t timestamp)
+uint64_t mxlTimestampToIndex(Rational const* editRate, uint64_t timestamp)
 {
     if ((editRate == nullptr) || (editRate->denominator == 0) || (editRate->numerator == 0))
     {
         return MXL_UNDEFINED_INDEX;
     }
 
-    return (timestamp * editRate->numerator + 500'000'000ULL * editRate->denominator) / (1'000'000'000ULL * editRate->denominator);
+    auto const nsPerGrain = (__uint128_t)editRate->denominator * (__uint128_t)1'000'000'000 / (__uint128_t)editRate->numerator;
+    return timestamp / (uint64_t)nsPerGrain;
```

Comment:
The idea behind switching to 128 bit integers was to avoid the overflow in the original calculation and avoid the inaccurate calculation via "nanoseconds per grain".

Also I would prefer C++ style casts.

So this would become:
```cpp
// ...
return static_cast<uint64_t>(timestamp * __uint128_t{editRate->numerator} + 500'000'000U * __uint128_t{editRate->denominator}) / (1'000'000'000U * __uint128_t{editRate->denominator});
```